st2actions/st2actions/runners/httprunner.py

```diff
@@ -171,19 +173,36 @@ def run(self):
         results = {}
         resp = None
         try:
-            resp = requests.request(
-                self.method,
-                self.url,
-                params=self.params,
-                data=self.body,
-                headers=self.headers,
-                cookies=self.cookies,
-                auth=self.auth,
-                timeout=self.timeout,
-                allow_redirects=self.allow_redirects,
-                proxies=self.proxies,
-                files=self.files
-            )
+            if self._is_json_content():
+                # use the json property instead of data in case of json content
+                # also cast property to a dict.
+                resp = requests.request(
+                    self.method,
+                    self.url,
+                    params=self.params,
+                    json=self._cast_object(self.body),
```

Comment:
This look good to me, although I would personally find it a bit more clear and readable if we explicitly convert it to json ourselves, e.g., something like:

```python
if json_content:
    data = self._cast_object(self.body)

    try:
        data = json.loads(data_string)
    except ValueError:
       // Complain loudly what failed
       raise Exception('Request body (%s), contain's invalid JSON')
else:
    data = self.body
```

This way, we also need only one call to reqiests since all the arguments are the same.